### DISCUSSION TITLE
Can't push to GitHub

### DISCUSSION THREAD

**Discussion by ReggieMoto** [community/community#21982] [New to GitHub]
<p>I’ve been using GitHub for a few years, although not for a little while until recently. I see now GitHub is deploying/has deployed 2FA and I have registered. That works. Today, though, I’ve found I cannot push my changes to origin: The requested URL returned error: 403</p>
<p>I’ve investigated further and found I needed a Personal Access Token (PAT). So I generated one for fetching and pulling from the remote. Still getting the 403.</p>
<p>I did some more investigating and found I maybe should be inserting my username as part of the GitHub URL: remote.origin.url=https://ReggieMoto@github.com/ReggieMoto/my-repo.git. Tried that and still getting the 403.</p>
<p>I’m certain this is all tied into the 2FA and the PAT somehow. I have all my credentials and am using them correctly, as far as I can tell. However:<br>
git push origin my-repo-branch<br>
Password for ‘<a href="https://ReggieMoto@github.com" rel="noopener nofollow ugc">https://ReggieMoto@github.com</a>’:<br>
remote: Permission to ReggieMoto/my-repo.git denied to ReggieMoto.<br>
fatal: unable to access ‘<a href="https://github.com/ReggieMoto/my-repo.git/" rel="noopener nofollow ugc">https://github.com/ReggieMoto/my-repo.git/</a>’: The requested URL returned error: 403</p>
<p>So, what’s going wrong?</p>
<p>ReggieMoto</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Using the PAT instead of your password when pushing should be the only change you need. Does your token have the right scopes? For pushing you’ll definitely need the <code>repo</code> scope, and <code>workflow</code> to add change Actions workflows.</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/developers/apps/scopes-for-oauth-apps#available-scopes" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/free-pro-team@latest/developers/apps/scopes-for-oauth-apps#available-scopes" target="_blank" rel="noopener">Scopes for OAuth Apps - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/free-pro-team@latest/developers/apps/scopes-for-oauth-apps#available-scopes" target="_blank" rel="noopener">//docs.github.com/en/free-pro-team@latest/developers/apps/scopes-for-oauth-apps</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by kingthorin** [comment]
<p>Your remote origin spec format is wrong when using https there’s no user@.</p>

**Comment by ReggieMoto** [comment]
<p>Many thanks, airtower-luna. The missing component was the workflow.</p>

**Comment by ReggieMoto** [comment]
<p>Seems to work fine. It just doesn’t prompt me for my username since it’s already embedded within the URL. The problem was the missing workflow scope from the PAT</p>

**Comment by macloo** [comment]
<p>“Does your token have the right scopes? For pushing you’ll definitely need the repo scope …” OMG it took me SO LONG to find this one simple fact, and it made everything work.</p>
<p>When you create your personal access token, click the box for repo — then use the token instead of password when you push, and it all works normally!</p>
<p>Thank you, @<a href="https://github.community/u/airtower-luna">airtower-luna</a>!!!</p>
<p>.</p>
